### PR TITLE
Aha/scheduler

### PR DESCRIPTION
**Pull request by andyHa**

### CODE REVIEW

File: src/main/java/sirius/biz/web/BasePageHelper.java [hunk 1]
```diff
@@ -192,6 +195,45 @@ public B addEnumFacet(String name, String title, Class<? extends Enum<?>> enumTy
...
+                               @Nonnull String title,
+                               @Nullable ValueComputer<String, String> translator) {
+        Objects.requireNonNull(ctx);
+
+        String value = ctx.get(parameterName).asString();
+
+        if (Strings.isFilled(value)) {
+            Facet facet = new Facet(title, parameterName, value, translator);
+            Value labelAsValue = ctx.get(labelParameterName);
+            facet.addItem(value, labelAsValue.getString(), -1);
```
Comment (sabieber) [inline src/main/java/sirius/biz/web/BasePageHelper.java]: Is this intentional possibly null? Below `asString` is used on the same value to create the second `Facet`. If this is by mistake the Value above could be replaced by a direct call to `asString` or `getString` and `Strings.isFilled` could be used in the `if`

Comment (andyHa) [reply src/main/java/sirius/biz/web/BasePageHelper.java]: if value is null, the parameter is hidden - thats the whole point of this facet. The label can also be null, then the translator will be used (if available). If both are null, the facet is ignored again.

File: src/main/java/sirius/biz/jobs/scheduler/JobConfigData.java [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+import sirius.kernel.health.HandledException;
+import sirius.kernel.health.Log;
+import sirius.web.http.WebContext;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Function;
+
+/**
+ * Represents a composire which can be embedded into an {@link sirius.db.mixing.BaseEntity} and contain all relevent
```
Comment (sabieber) [inline src/main/java/sirius/biz/jobs/scheduler/JobConfigData.java]:
```suggestion
 * Represents a composite which can be embedded into a {@link sirius.db.mixing.BaseEntity} and contain all relevent
```

File: src/main/java/sirius/biz/jobs/scheduler/JobConfigData.java [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+ * data to describe a job a its start parameters.
```
Comment (sabieber) [inline src/main/java/sirius/biz/jobs/scheduler/JobConfigData.java]:
```suggestion
 * data to describe a job and its start parameters.
```

File: src/main/java/sirius/biz/jobs/scheduler/JobConfigData.java [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+
+    /**
+     * Contains the name of the {@link JobFactory} to launch.
+     */
+    public static final Mapping JOB = Mapping.named("job");
+    @Length(255)
+    private String job;
+
+    /**
+     * Contains the a copy of {@link JobFactory#getLabel()} to quickly render the selected job.
```
Comment (sabieber) [inline src/main/java/sirius/biz/jobs/scheduler/JobConfigData.java]:
```suggestion
     * Contains a copy of {@link JobFactory#getLabel()} to quickly render the selected job.
```

File: src/main/java/sirius/biz/jobs/scheduler/JobConfigData.java [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+                JSON.parseObject(configuration)
+                    .forEach((key, value) -> configMap.put(key, value == null ? null : value.toString()));
+            }
+        }
+
+        return configMap;
+    }
+
+    /**
+     * Returns a parameter provide to be supplied to {@link JobFactory#startInBackground(Function)}.
```
Comment (sabieber) [inline src/main/java/sirius/biz/jobs/scheduler/JobConfigData.java]:
```suggestion
     * Returns a parameter provider to be supplied to {@link JobFactory#startInBackground(Function)}.
```

File: src/main/java/sirius/biz/jobs/scheduler/SchedulerController.java [hunk 6]
```diff
@@ -0,0 +1,183 @@
...
+
+    /**
+     * Lists all scheduler entries for the current tenant.
+     *
+     * @param ctx the current request
+     */
+    @Permission(PERMISSION_MANAGE_SCHEDULER)
+    @Routed("/jobs/scheduler")
+    public void schedulerEntries(WebContext ctx) {
+        BasePageHelper<J, ?, ?, ?> ph = getEntriesAsPage();
```
Comment (sabieber) [inline src/main/java/sirius/biz/jobs/scheduler/SchedulerController.java]: ph -> pageHelper

File: src/main/java/sirius/biz/jobs/scheduler/SchedulerData.java [hunk 7]
```diff
@@ -0,0 +1,337 @@
...
+import sirius.kernel.nls.NLS;
+
+import java.time.Duration;
+import java.time.LocalDateTime;
+import java.util.function.Consumer;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Contains a composite which describe the scheduling setup of an {@link SchedulerEntry}.
```
Comment (sabieber) [inline src/main/java/sirius/biz/jobs/scheduler/SchedulerData.java]:
```suggestion
 * Contains a composite which describes the scheduling setup of an {@link SchedulerEntry}.
```

File: src/main/java/sirius/biz/jobs/scheduler/SchedulerData.java [hunk 8]
```diff
@@ -0,0 +1,337 @@
...
+     * <p>
+     * Note that this will only fill field and not persist anything to the database.
+     *
+     * @param timestamp the time when the task was executed
+     */
+    public void rememberExecution(LocalDateTime timestamp) {
+        this.lastExecution = timestamp;
+        this.numberOfExecutions++;
+        if (this.runs != null) {
+            this.runs = this.runs - 1;
```
Comment (sabieber) [inline src/main/java/sirius/biz/jobs/scheduler/SchedulerData.java]:
```suggestion
            this.runs -= 1;
```

**Pull request merged**